libevmasm/ConstantOptimiser.cpp

```diff
@@ -53,17 +54,22 @@ unsigned ConstantOptimisationMethod::optimiseConstants(
 		bigint copyGas = copy.gasNeeded();
 		ComputeMethod compute(params, item.data());
 		bigint computeGas = compute.gasNeeded();
+		AssemblyItems replacement;
 		if (copyGas < literalGas && copyGas < computeGas)
 		{
-			copy.execute(_assembly, _items);
+			replacement = copy.execute(_assembly);
 			optimisations++;
 		}
 		else if (computeGas < literalGas && computeGas < copyGas)
```

Comment:
The last comparison should be `<=`.  Otherwise, when `computeGas == copyGas`, no optimizations happen.